pkg/controller/servicebindingrequest/retriever.go

```diff
@@ -28,6 +28,24 @@ const (
 	secretPrefix  = "urn:alm:descriptor:servicebindingrequest:env:object:secret"
 )
 
+func getNestedValue(key string, sectionMap interface{}) (string, error) {
+	var value interface{}
+	var exists bool
+	if !strings.Contains(key, ".") {
+		value, exists = sectionMap.(map[string]interface{})[key]
+		if !exists {
+			return "", fmt.Errorf("Can't find key '%s'", key)
+		}
+		return fmt.Sprintf("%v", value), nil
+	}
+	attrs := strings.SplitN(key, ".", 2)
```

Comment:
Do you need to check for `len()`, before doing the `SplitN`?

Reply:
That's not really required. This function will not return any error.